### ISSUE TITLE
123

### ISSUE THREAD

**Issue by UnlcleTian** [UnlcleTian/kaggle#2]
import numpy as np
import pandas as pd
# from sklearn.model_selection import KFold
from lightgbm import LGBMClassifier

import plotly.offline as py
import plotly.graph_objs as go
import matplotlib.pyplot as plt
import seaborn as sns
import gc

data_train = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\train.csv')

data_test = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\test.csv')
# data_his_tra = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\historical_transactions.csv')

# data_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\merchants.csv')

data_new_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\new_merchant_transactions.csv')

print('new shape:',data_new_mer.shape)
# print('type:',data_new_mer.dtypes)

# count the null cell??
# print('null:',data_new_mer.isnull())

# A:2,B:3,?

data_new_mer_dum1 = pd.get_dummies(data_new_mer.category_1,prefix='cate1',dummy_na=True)
data_new_mer_dum2 = pd.get_dummies(data_new_mer.category_3,prefix='cate3',dummy_na=True)

print(data_new_mer_dum1.head())
print(data_new_mer_dum1.shape)

'''
plot picture for discribution of data column

折れ線グラフ / 棒グラフを一つのプロットとして描画する場合は以下のようにする。.plot メソッドは matplotlib.axes.Axes インスタンスを返すため、続くプロットの描画先として その Axes を指定すればよい。
ax = monthly[u'札幌'].plot(legend=True)
monthly[[u'東京', u'福岡']].plot.bar(ax=ax, rot=30)
&&&&&&&&&&&&&&&
plotの戻り値としてaxesが返ってくるので、それを指定すればまとめられる。 

試しに2つのscatterを書いてみる。 
# 1つめを青でplot
df1 = pd.DataFrame( np.random.randn(100, 2), columns=('A', 'B') )
ax = df1.plot( kind='scatter', x='A', y='B', color='blue' )

# 2つめを緑でplot(ax指定)
df2 = pd.DataFrame( np.random.randn(100, 2) + 1, columns=('A', 'B') ) 
df2.plot( kind='scatter', x='A', y='B', color='green', ax = ax )

&&&&&&&&&&&&&6

city_id,merchant_category_id
state id vs score
'''

data_new_mer = pd.concat([data_new_mer,data_new_mer_dum1,data_new_mer_dum2],axis=1 ).drop(['category_1','category_3'],axis=1)

print('&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&')
# print(data_new_mer.info(null_counts=True))

'''sns.violinplot(x=)'''

'''check distribution of data'''
# fig,ax = plt.subplots(1,3,figsize=(16,6))

# data_new_mer['cate1_N'].plot(kind='hist',ax=ax[0],title='cate1_N')

# plt.suptitle('77777')
# print(data_new_mer.dtypes)

# print(data_new_mer['city_id'].value_counts(normalize=True))

# data_new_mer['city_id'].plot(kind='hist',ax=ax[1],title='city_id')

# plt.show()

# plt.scatter(x= data_new_mer['cate1_N'].value_counts(1).index,y= data_new_mer['cate1_N'].value_counts(1).values,)

# plt.scatter(x= data_new_mer['cate1_Y'].value_counts(1).index,y= data_new_mer['cate1_Y'].value_counts(1).values,)

# plt.show()
'''
fig,ax =plt.subplot(1,3,figsize=(16,6))
plt.suptitle('wow')
sns.violinplot(x='feature1',y='target',data=train,ax=ax[0],title= 'feature1')
##############################

d1 = train['first_active_month'].value_counts().sort_index()
d2 = test['first_active_month'].vaule_counts().sort_index()

data = [go.Scatter(x=d1.index,y=d1.values, name= train),
    go.Scatter(x= d2.index,y=d2.values, name=test)]
layout= go.layout(dict(title= 'counts',xaxis=dict(title='month'),
    yaxis= dict(title= 'count'),
    legend= dict(orientation='v'))
py.iplot(dict(data=data,layout =layout))

'''
print(data_new_mer.head())

diff_train= list(set(data_test['card_id']).difference(set(data_train['card_id'])))
diff_new_mer = list(set(data_test['card_id']).difference(set(data_new_mer['card_id'])))

diff_train_newMer = list(set(data_train['card_id']).difference(set(data_new_mer['card_id'])))

print(data_test.shape)
print(data_train.shape)
print(data_new_mer.shape)

print(np.count_nonzero(diff_train))
print(np.count_nonzero(diff_new_mer))
print(np.count_nonzero(diff_train_newMer))

# result = pd.DataFrame(data_new_mer['card_id'].sort_values(ascending= True).unique() == data_test['card_id'].sort_values(ascending= True).unique() )

# print(result.value_counts())

**Comment by UnlcleTian** [comment]
import numpy as np
import pandas as pd
# from sklearn.model_selection import KFold
from lightgbm import LGBMClassifier

import plotly.offline as py
import plotly.graph_objs as go
import matplotlib.pyplot as plt
import seaborn as sns
import gc

data_train = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\train.csv')

data_test = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\test.csv')
# data_his_tra = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\historical_transactions.csv')

# data_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\merchants.csv')

data_new_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\new_merchant_transactions.csv')

print('new shape:',data_new_mer.shape)
# print('type:',data_new_mer.dtypes)

# count the null cell??
# print('null:',data_new_mer.isnull())

# A:2,B:3,?

data_new_mer_dum1 = pd.get_dummies(data_new_mer.category_1,prefix='cate1',dummy_na=True)
data_new_mer_dum2 = pd.get_dummies(data_new_mer.category_3,prefix='cate3',dummy_na=True)

print(data_new_mer_dum1.head())
print(data_new_mer_dum1.shape)

'''
plot picture for discribution of data column


折れ線グラフ / 棒グラフを一つのプロットとして描画する場合は以下のようにする。.plot メソッドは matplotlib.axes.Axes インスタンスを返すため、続くプロットの描画先として その Axes を指定すればよい。
ax = monthly[u'札幌'].plot(legend=True)
monthly[[u'東京', u'福岡']].plot.bar(ax=ax, rot=30)
&&&&&&&&&&&&&&&
plotの戻り値としてaxesが返ってくるので、それを指定すればまとめられる。 

試しに2つのscatterを書いてみる。 
# 1つめを青でplot
df1 = pd.DataFrame( np.random.randn(100, 2), columns=('A', 'B') )
ax = df1.plot( kind='scatter', x='A', y='B', color='blue' )

# 2つめを緑でplot(ax指定)
df2 = pd.DataFrame( np.random.randn(100, 2) + 1, columns=('A', 'B') ) 
df2.plot( kind='scatter', x='A', y='B', color='green', ax = ax )

&&&&&&&&&&&&&6


city_id,merchant_category_id
state id vs score
'''

data_new_mer = pd.concat([data_new_mer,data_new_mer_dum1,data_new_mer_dum2],axis=1 ).drop(['category_1','category_3'],axis=1)

print('&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&')
# print(data_new_mer.info(null_counts=True))

'''sns.violinplot(x=)'''

'''check distribution of data'''
# fig,ax = plt.subplots(1,3,figsize=(16,6))

# data_new_mer['cate1_N'].plot(kind='hist',ax=ax[0],title='cate1_N')

# plt.suptitle('77777')
# print(data_new_mer.dtypes)


# print(data_new_mer['city_id'].value_counts(normalize=True))

# data_new_mer['city_id'].plot(kind='hist',ax=ax[1],title='city_id')

# plt.show()

# plt.scatter(x= data_new_mer['cate1_N'].value_counts(1).index,y= data_new_mer['cate1_N'].value_counts(1).values,)

# plt.scatter(x= data_new_mer['cate1_Y'].value_counts(1).index,y= data_new_mer['cate1_Y'].value_counts(1).values,)

# plt.show()
'''
fig,ax =plt.subplot(1,3,figsize=(16,6))
plt.suptitle('wow')
sns.violinplot(x='feature1',y='target',data=train,ax=ax[0],title= 'feature1')
##############################


d1 = train['first_active_month'].value_counts().sort_index()
d2 = test['first_active_month'].vaule_counts().sort_index()

data = [go.Scatter(x=d1.index,y=d1.values, name= train),
    go.Scatter(x= d2.index,y=d2.values, name=test)]
layout= go.layout(dict(title= 'counts',xaxis=dict(title='month'),
    yaxis= dict(title= 'count'),
    legend= dict(orientation='v'))
py.iplot(dict(data=data,layout =layout))

'''
print(data_new_mer.head())


'''check consistency '''
# diff_train= list(set(data_test['card_id']).difference(set(data_train['card_id'])))
# '''
# former minuns latter one
# '''
# diff_new_mer = list(set(data_test['card_id']).difference(set(data_new_mer['card_id'])))

# diff_train_newMer = list(set(data_train['card_id']).difference(set(data_new_mer['card_id'])))

# print(data_test.shape)
# print(data_train.shape)
# print(data_new_mer.shape)

# print(np.count_nonzero(diff_train))
# print(np.count_nonzero(diff_new_mer))
# print(np.count_nonzero(diff_train_newMer))


print(data_new_mer.dtypes)
#print(data_new_mer.describe(include = 'all'))

print(data_new_mer.select_dtypes(include='object'))

data_new_mer['purchase_date'] = pd.to_datetime(data_new_mer['purchase_date'])


# print(data_new_mer.purchase_date.dt.dayofweek)
# print(data_new_mer.purchase_date.dt.quarter)
# fig,ax = plt.subplots(2,3,figsize=(16,6))
# data_new_mer.purchase_date.dt.dayofweek.value_conuts().plot(kind= 'hist',ax=ax[1,1],title= 'week',align= 'left')
# data_new_mer.purchase_date.dt.quarter.plot(kind= 'hist', ax = ax[0,2],title = 'quarter',align= 'mid',color='red')
# plt.suptitle('EDA')
# plt.show()
#fig.savefig(r'C:\Users\guanhua.wang\Downloads\data\pandas_iris_line_axes.png')
'''bar '''
# fig,ax = plt.subplots(1,2,figsize=(12,4))
# data_new_mer.purchase_date.dt.dayofweek.value_counts().sort_index().plot(kind= 'bar',ax=ax[0],title= 'week',color='teal')
# data_new_mer.purchase_date.dt.quarter.value_counts().sort_index().plot(kind= 'bar', ax = ax[1],title = 'quarter',color='gold')
# plt.suptitle('EDA')
# plt.show()
# fig.savefig(r'C:\Users\guanhua.wang\Downloads\data\pandas_iris_line_axes2.png')
